website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,350 @@
+# Apache Hudi Storage Format Specification [DRAFT]
+
+
+
+This document is a specification for the Hudi Storage Format which transforms immutable cloud/file storage systems into transactional data lakes. 
+
+## Overview
+
+Hudi Storage Format enables the following features over very large collection of files/objects
+
+- streaming primitives like incremental merges, change stream etc
+- database primitives like tables, transactions, mutability, indexes and query performance optimizations 
+
+Apache Hudi is an open source data lake platform that is built on top of the Hudi Storage Format and it unlocks the following features 
```

Comment:
Apart from features listed below, do we also want to mention design principles? https://cwiki.apache.org/confluence/display/HUDI/Design+And+Architecture#DesignAndArchitecture-DesignPrinciples

Reply:
Yeah I think we have emphasized relevant aspects of design principles in the storage format here. Anything specific you want to add on here?